src/Fractal.php

```diff
@@ -454,8 +454,11 @@ public function getResourceName()
 
     /**
      * Convert the object into something JSON serializable.
+    +     *
+    +     * @return array
      */
-    public function jsonSerialize(): mixed
+    #[\ReturnTypeWillChange]
+    public function jsonSerialize()
```

Comment:
Could we add the right return type hint instead of using the attribute?

Reply:
@freekmurze 
I changed it to return `array`, but the attribute may be needed as well to prevent a `deprecated` warning in PHP 8.1.

`Fractal::jsonSerialize()` implements this interface, where return type is `mixed`

```php
// Source: json.php
interface JsonSerializable
{
    /**
     * Specify data which should be serialized to JSON
     * @link https://php.net/manual/en/jsonserializable.jsonserialize.php
     * @return mixed data which can be serialized by <b>json_encode</b>,
     * which is a value of any type other than a resource.
     * @since 5.4
     */
    public function jsonSerialize();
}
```

`Fractal::jsonSerialize()` is calling `Fractal::toArray()`, that returns an `array`

```php
// Source: src/Fractal.php
class Fractal implements JsonSerializable
    /**
     * Perform the transformation to array.
     *
     * @return array
     */
    public function toArray()
    {
        return $this->createData()->toArray();
    }
}
```

```php
// Source: vendor/league/fractal/src/Scope.php
class Scope implements \JsonSerializable {
    public function toArray(): ?array
    {
       // ...
    }
}
```

---

By the way, I can see other deprecated warnings for other methods, when running unit tests with PHP 8.1, for the same reason. To be fixed separately.

Reply:
Feel free to PR other needed fixes 👍